RoadNamer/Managers/RoadRenderingManager.cs

```diff
@@ -111,25 +113,25 @@ private void RenderText()
                                 NetNode endNode = netManager.m_nodes.m_buffer[netSegment.m_endNode];
 
                                 Vector3 segmentLocation = netSegment.m_bounds.center;
-
-                                float scaleMultiplier = m_textQuality / 20f;
-
                                 road.m_textMesh.anchor = TextAnchor.MiddleCenter;
                                 road.m_textMesh.font = districtManager.m_properties.m_areaNameFont.baseFont;
                                 road.m_textMesh.GetComponent<Renderer>().material = road.m_textMesh.font.material;
+                                road.m_textMesh.GetComponent<Renderer>().receiveShadows = true;
                                 road.m_textMesh.fontSize = (int)Math.Round(m_textQuality);
                                 road.m_textMesh.transform.position = startNode.m_position;
                                 road.m_textMesh.transform.LookAt(endNode.m_position, Vector3.up);
+                                Vector3 rotation = Vector3.Cross(startNode.m_position, endNode.m_position);
                                 road.m_textMesh.transform.Rotate(90f, 0f, 90f);
```

Comment:
It may be possible to check for rotation based on the segment flags ( there's an inverted flag, and it seems to affect rendering )